### ISSUE TITLE
cabal2nix fails on cabal://postgresql-simple

### ISSUE THREAD

**Issue by chris-martin** [NixOS/cabal2nix#501]
```
$ cabal2nix --version
cabal2nix 2.17.0

$ cabal2nix cabal://postgresql-simple
cabal2nix: HackageDBPackageName (PackageName "postgresql-simple") (InvalidRepresentationOfType "preferred version range" "postgresql-simple <0.4.1.0 || >0.4.1.0 && <0.4.2.0 || >0.4.2.0 && <0.4.3.0 || >0.4.3.0 && <0.4.4.0 || >0.4.4.0 && <0.5.0.0 || >0.5.0.0")
```

**Comment by chris-martin** [comment]
The error only happens when I get cabal2nix from the unstable nixos channel. It works in nixos-21.05, which is still 2.17.0. So it must be caused by an update to some dependency?

**Comment by sternenseemann** [comment]
Unstable and 21.05 have the same `cabal2nix` version (2.17.0), so I'm suspecting a dependency update may be responsible for this, possibly?

**Comment by sternenseemann** [comment]
Okay, I've confirmed that the culprit is `hackage-db` 2.1.1 which included a [fix to compatibility with Cabal 3.4](https://github.com/peti/hackage-db/pull/12). Seems like that update should also have increased the lower version bound on Cabal.

I'll downgrade `hackage-db` in nixpkgs for the compilers that don't have Cabal 3.4 later which should land in `master` this weekend hopefully. Currently testing `hackage-db` with GHC 9.0.1 to confirm that this issue doesn't happen with Cabal 3.4.

**Comment by sternenseemann** [comment]
Nevermind, `cabal2nix` doesn't support Cabal 3.4 yet. Can you try checking out <https://github.com/nixos/nixpkgs/commit/c145f7930df909d583fe6d7fc862ff963a095f7a> and trying this again? I can't reproduce the error with that change anymore.

**Comment by sternenseemann** [comment]
The issue persists with Cabal 3.4.0.0 and hackage-db 2.1.1 (tested with #503).

**Comment by sternenseemann** [comment]
Building cabal2nix against the just released [hackage-db 2.1.2](https://hackage.haskell.org/package/hackage-db-2.1.2) resolves this issue. I expect this to be fixed in nixpkgs within a week or two as well.

Let me know if you have any further problems.
